tracer/src/Datadog.Trace/Iast/Aspects/System.Runtime/DefaultInterpolatedStringHandlerAspect.cs

```diff
@@ -0,0 +1,155 @@
+// <copyright file="DefaultInterpolatedStringHandlerAspect.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if NET6_0_OR_GREATER
+
+using System;
+using System.Reflection.Emit;
+using System.Runtime.CompilerServices;
+using Datadog.Trace.Iast.Dataflow;
+using Datadog.Trace.Iast.Propagation;
+using InlineIL;
+using static InlineIL.IL.Emit;
+
+#nullable enable
+
+namespace Datadog.Trace.Iast.Aspects.System.Runtime;
+
+/// <summary> DefaultInterpolatedString class aspect </summary>
+[AspectClass("System.Runtime")]
+[global::System.ComponentModel.Browsable(false)]
+[global::System.ComponentModel.EditorBrowsable(global::System.ComponentModel.EditorBrowsableState.Never)]
+public class DefaultInterpolatedStringHandlerAspect
+{
+    /// <summary>
+    /// System.Runtime DefaultInterpolatedStringHandler.AppendFormatted(String) aspect
+    /// </summary>
+    /// <param name="target"> the ref DefaultInterpolatedStringHandler </param>
+    /// <param name="value"> the string value </param>
+    [AspectMethodReplace("System.Runtime.CompilerServices.DefaultInterpolatedStringHandler::AppendFormatted(System.String)")]
+    public static void AppendFormatted1(ref DefaultInterpolatedStringHandler target, string value)
+    {
+        target.AppendFormatted(value);
+        try
+        {
+            DefaultInterpolatedStringHandlerModuleImpl.Append(ToPointer(ref target), value);
+        }
+        catch (Exception ex)
+        {
+            IastModule.Log.Error(ex, $"Error invoking {nameof(DefaultInterpolatedStringHandlerAspect)}.{nameof(AppendFormatted1)}");
```

Comment:
_Consider_ using `Warning` here, or gating this so you only report the error once. Otherwise, if there's an issue with `ToPointer` (for example) then you'll flood the customer's logs with errors...

Reply:
Applied in e777c49 😄